src/ConnectedMode/ServerSentEvents/SSESessionManager.cs

```diff
@@ -67,25 +71,42 @@ public void Dispose()
 
         private void SolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs activeSolutionBindingEventArgs)
         {
-            EndCurrentSession();
-
-            var bindingConfiguration = activeSolutionBindingEventArgs.Configuration;
-            var isInConnectedMode = !bindingConfiguration.Equals(BindingConfiguration.Standalone);
+            CreateSessionIfInConnectedMode(activeSolutionBindingEventArgs.Configuration);
+        }
 
-            if (!isInConnectedMode)
+        private void CreateSessionIfInConnectedMode(BindingConfiguration bindingConfiguration)
+        {
+            ISSESession sessionToLaunch;
+            lock (syncRoot)
             {
-                return;
-            }
+                if (bindingConfiguration.Equals(currentBindingConfiguration))
+                {
+                    return;
+                }
 
-            currentSession = sseSessionFactory.Create(bindingConfiguration.Project.ProjectKey);
+                EndCurrentSession();
 
-            currentSession.PumpAllAsync().Forget();
+                currentBindingConfiguration = bindingConfiguration;
```

Comment:
I think we should not cache this.
This solves a scenario that can only happen during first time initialization, and it is unlikely to happen in practice. If we don't cache, then worst case we will throw away the session and make a new one which is identical.
I think this is preferred as it simplifies the code and the impact is not large.

Reply:
Edit: I think it's also dangerous. The user can force an update by clicking on "Update" in the team explorer. I _think_ that doing that fires this event too. This will kill the force-update functionality.

Reply:
Will the active solution tracker even raise an event in this case? Since it's the same binding configuration, I'm pretty sure it's not going to

Reply:
You can check; put a breakpoint in the event and click on update.